Source/ImPlotWidgets/Private/HistogramWidget.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) Coconut Lizard Limited. All rights reserved.
+
+#include "ImPlotWidgets/Public/HistogramWidget.h"
+
+#include <ImGui/Public/ImPlotImplementation.h>
+#include <Runtime/Core/Public/Algo/Sort.h>
+
+void SHistogramWidget::Construct(const FArguments& InArgs)
+{
+	SImPlotWidget::Construct(InArgs);
+}
+
+void SHistogramWidget::OnImGuiDraw() const
+{
+	SImPlotWidget::OnImGuiDraw();
+
+	const FGeometry& AllottedGeometry = ImGuiCustomWidget.Get()->GetTickSpaceGeometry();
+
+	constexpr int32 GraphFlags = ImPlotFlags_AntiAliased;
+	constexpr int32 AxisFlags = ImPlotAxisFlags_NoGridLines;
+
+    // ---------------------------------
+    {
+		static int bins = 50;
```

Comment:
If there's multiple instances of the histogram this will affect all of them, so basically they will all have the same number of bins and all sliders will change it for all of them, fixing this atm

Reply:
I think the number of bins should probably be based on the number of data points passed in really, and should definitely be different for each instance of the histogram. I know that the const makes this annoying but I think we should make this a mutable.

Reply:
https://en.wikipedia.org/wiki/Histogram#Number_of_bins_and_width
Worth reading some of this about a good default value for the bins, that can be changed if the user wants, maybe?
Also worth looking into whether the ImPlot histogram allows bars of _different_ width: https://en.wikipedia.org/wiki/Histogram#Variable_bin_widths
A little bit overly mathsy so let me know if I can clarify anything